helper.php

```diff
@@ -0,0 +1,336 @@
+			'parameters' => array(
+				'folder' => "string path of to look up (e.g. ':mynamespace:mypage'), if null use current page.",
+			),
+			'return' => "array the config settings key=>value array, e.g. array('pagesize'=>'A4', 'orientation'=>'protrait' ...)",
+		);
+		$result[] = array(
+			'name'   => 'loadExportConfig',
+			'desc'   => "Returns an array of config settings that are going to be used. This includes checking $INPUT for overrides and using current page's hierarchical settings if enabled.",
+			'return' => "array the config settings key=>value array, e.g. array('pagesize'=>'A4', 'orientation'=>'protrait' ...)",
+		);
+		$result[] = array(
+			'name'   => 'addExportButton',
+			'desc'   => "Adds the export by (hierarchical) settings.",
+			'return' => "void",
+		);
+		$result[] = array(
+			'name'   => 'replyAjax',
+			'desc'   => "Handles calls.",
+			'return' => "void (it handled the call so nothing more to do)",
+		);
+		return $result;
+	}
+	
+	function checkSettingstree(){
+		return !!$this->get_settingstree();
+	}
+
+	function getSettingstreeAdminHtml(){
+		if (!$this->get_settingstree()){ return null;}
+		return $this->settings_helper->showAdmin('dw2pdf',$this->_getFolder(null));
+	}
+	private function _getFolder($folder = null){
+		if (is_null($folder)){
+			global $ID;
+			$folder = $ID;
+		}
+		return ':'.ltrim(strtr($folder,'/',':'),':');
+	}
+
+	function getConf($key,$notset = false, $folder = null){
+		$folder = $this->_getFolder($folder);
+		if (!is_array(@$this->_folder_confs[$folder])){
+			$this->getAllConf($folder);
+		}
+		
+		return (array_key_exists($key, (array)$this->_folder_confs[$folder]) ? $this->_folder_confs[$folder][$key] : $notset);
+	}
+	
+	function getAllConf($folder = null){
+		$folder = $this->_getFolder($folder);
+		if (!is_array(@$this->_folder_confs[$folder])){
+			global $conf;
+			$cnf = $conf['plugin']['dw2pdf'];
+			if ($this->get_settingstree()){
+				$cnf = array_merge($cnf,$this->settings_helper->getConf('dw2pdf',$folder));
+			}
+			if (!is_array($cnf)) {$cnf = array();}	// should never happen if the plugin have configs, and this plugin have configs.
+			$this->_folder_confs[$folder] = $cnf;
+		}
+		return $this->_folder_confs[$folder];
+
+	}
+	
+	// CODE MOVED HERE FROM action.php
+	function addExportButton(&$event){
+        if($this->getConf('showexportbutton') && $event->data['view'] == 'main') {
+			// insert button at position before last (up to top)
+			$event->data['items'] = array_slice($event->data['items'], 0, -1, true) +
+				array('export_pdf' => 
+						  '<li>'
+						  .'<a ' . $this->_exportButtonHtml()	// this is the htmlcode for the opening a tag's href and onclick attributes.
+						  . '"  class="action export_pdf" rel="nofollow" title="' . $this->getLang('export_pdf_button') . '">'
+						  . '<span>' . $this->getLang('export_pdf_button') . '</span>'
+						  . '</a>'
+						  . '</li>'
+				) +
+				array_slice($event->data['items'], -1, 1, true);
```

Comment:
I moved it there, as it highly depend on environment (i.e. current ID if settingstree enabled), and that would be the helper's task to deal with.

Separate by functionality: maybe if the helper just returns the html of the button and in the action inserts it back to the actual event?
In this case `$event->data['view']` would be evaluated in action, and only `$this->getConf('showexportbutton') ` in the helper.